### PR TITLE
Add new test cases for new classes

### LINKED ISSUE #63: Add new test cases for new parameetrs
To add test cases for the following new classes:
- [ ] Nric
- [ ] Gender
- [ ] DateOfBirth
- [ ] DateUtil

### PR DESCRIPTION
**Pull request by yong-zhuo**
Closes #63 

Added test cases for the following class:
1. Nric
2. Gender
3. DateOFBirth
4. DateUtil

### CODE REVIEW

File: src/test/java/seedu/address/commons/util/DateUtilTest.java [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+    public void isAfterToday_dateBeforeToday_returnsFalse() {
+        String today = LocalDate.now().toString();
+        assertFalse(isAfterToday("2000-01-01")); // past date
+        assertFalse(isAfterToday(today)); // today's date or past date
+    }
+
+    @Test
+    public void isAfterToday_invalidDate_throwsIllegalArgumentException() {
+        assertThrows(IllegalArgumentException.class, () -> isAfterToday("invalid-date"));
+    }
```
Comment (PresleyChew) [inline src/test/java/seedu/address/commons/util/DateUtilTest.java:46]: Comprehensive testing, LGTM 👍🏻

File: src/test/java/seedu/address/model/person/DateOfBirthTest.java [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+    @Test
+    public void constructor_null_throwsNullPointerException() {
+        assertThrows(NullPointerException.class, () -> new DateOfBirth(null));
+    }
+
+    @Test
+    public void constructor_invalidDateOfBirth_throwsIllegalArgumentException() {
+        String invalidDateOfBirth = "invalid-date";
+        assertThrows(IllegalArgumentException.class, () -> new DateOfBirth(invalidDateOfBirth));
+    }
```
Comment (PresleyChew) [inline src/test/java/seedu/address/model/person/DateOfBirthTest.java:22]: Good test cases for testing constructor!

File: src/test/java/seedu/address/model/person/GenderTest.java [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+        //invalid Gender
+        assertFalse(Gender.isValidGender("")); // empty string
+        assertFalse(Gender.isValidGender(" ")); // spaces only
+        assertFalse(Gender.isValidGender("T")); // only invalid characters
+
+        //valid Gender
+        assertTrue(Gender.isValidGender("M")); // for Male
+        assertTrue(Gender.isValidGender("F")); //for Female
+        assertTrue(Gender.isValidGender("m")); //for lowercase representation
+        assertTrue(Gender.isValidGender("f")); //for lowercase representation
```
Comment (PresleyChew) [inline src/test/java/seedu/address/model/person/GenderTest.java:36]: Good test cases for uppercase and lowercase

File: src/test/java/seedu/address/model/person/NricTest.java [hunk 4]
```diff
@@ -0,0 +1,64 @@
...
+        assertFalse(Nric.isValidNric("^")); // only invalid characters
+        assertFalse(Nric.isValidNric("Z1234567A")); // invalid first letter
+        assertFalse(Nric.isValidNric("S12345672")); // invalid last letter
+        assertFalse(Nric.isValidNric("S12345678A")); // invalid length
+
+        // valid nric
+        assertTrue(Nric.isValidNric("S1234567A")); // valid nric all uppercase
+        assertTrue(Nric.isValidNric("s1234567a")); // valid nric all lowercase
+        assertTrue(Nric.isValidNric("S1234567a")); // valid nric with lowercase for last letter
+        assertTrue(Nric.isValidNric("s1234567B")); // valid nric with lowercase first letter
```
Comment (PresleyChew) [inline src/test/java/seedu/address/model/person/NricTest.java:39]: Comprehensive test cases, great job 💯

**Review by PresleyChew — COMMENTED** [review]
LGTM

File: src/test/java/seedu/address/model/person/DateOfBirthTest.java [hunk 5]
```diff
@@ -0,0 +1,56 @@
...
+import static seedu.address.testutil.Assert.assertThrows;
+
+import java.time.LocalDate;
+
+import org.junit.jupiter.api.Test;
+
+public class DateOfBirthTest {
+
+    @Test
+    public void constructor_null_throwsNullPointerException() {
```
Comment (hweinian) [inline src/test/java/seedu/address/model/person/DateOfBirthTest.java:14]: Good job on testing null pointers.

File: src/test/java/seedu/address/commons/util/DateUtilTest.java (hunk 1 above)
Comment (hweinian) [reply src/test/java/seedu/address/commons/util/DateUtilTest.java:46]: I second this! Comprehensive test cases.

File: src/test/java/seedu/address/model/person/DateOfBirthTest.java [hunk 6]
```diff
@@ -0,0 +1,56 @@
...
+
+        // null -> returns false
+        assertFalse(dateOfBirth.equals(null));
+
+        // different types -> returns false
+        assertFalse(dateOfBirth.equals(5.0f));
+
+        // different values -> returns false
+        assertFalse(dateOfBirth.equals(new DateOfBirth("1999-01-01")));
+    }
```
Comment (hweinian) [inline src/test/java/seedu/address/model/person/DateOfBirthTest.java:55]: Good use of assertions

File: src/test/java/seedu/address/model/person/GenderTest.java (hunk 3 above)
Comment (hweinian) [reply src/test/java/seedu/address/model/person/GenderTest.java:36]: I agree! good job on checking edge cases

File: src/test/java/seedu/address/model/person/NricTest.java [hunk 7]
```diff
@@ -0,0 +1,64 @@
...
+        // null nric
+        assertThrows(NullPointerException.class, () -> Nric.isValidNric(null));
+
+        // invalid nric
+        assertFalse(Nric.isValidNric("")); // empty string
+        assertFalse(Nric.isValidNric(" ")); // spaces only
+        assertFalse(Nric.isValidNric("^")); // only invalid characters
+        assertFalse(Nric.isValidNric("Z1234567A")); // invalid first letter
+        assertFalse(Nric.isValidNric("S12345672")); // invalid last letter
+        assertFalse(Nric.isValidNric("S12345678A")); // invalid length
```
Comment (hweinian) [inline src/test/java/seedu/address/model/person/NricTest.java:33]: Extensive testing on different possible inputs.

**Review by hweinian — COMMENTED** [review]
Overall LGTM!
